src/core/benches/gather.rs

```diff
@@ -0,0 +1,68 @@
+#[macro_use]
+extern crate criterion;
+
+use std::fs::File;
+use std::io::BufReader;
+use std::path::PathBuf;
+
+use sourmash::index::calculate_gather_stats;
+use sourmash::signature::Signature;
+use sourmash::sketch::Sketch;
+
+use criterion::{black_box, Criterion};
+
+fn gather_stats_benchmarks(c: &mut Criterion) {
+    let mut filename = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    filename.push("../../tests/test-data/gather-abund/genome-s10.fa.gz.sig");
+    let file = File::open(filename).unwrap();
+    let reader = BufReader::new(file);
+    let mut sigs: Vec<Signature> = serde_json::from_reader(reader).expect("Loading error");
+    let orig_query = if let Sketch::MinHash(mh) = &sigs.swap_remove(0).sketches()[0] {
+        mh.clone()
+    } else {
+        unimplemented!()
+    };
+    let query = orig_query.clone();
+
+    let mut filename = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    filename.push("../../tests/test-data/gather-abund/genome-s11.fa.gz.sig");
+    let file = File::open(filename).unwrap();
+    let reader = BufReader::new(file);
+    let mut sigs: Vec<Signature> = serde_json::from_reader(reader).expect("Loading error");
+    let match_sig = &sigs.swap_remove(0);
+    let match_mh = if let Sketch::MinHash(mh) = &match_sig.sketches()[0] {
+        mh.clone()
+    } else {
+        unimplemented!()
+    };
+
+    let test_cases = vec![(false, false), (true, false), (false, true), (true, true)];
+
+    let mut group = c.benchmark_group("gather_stats");
+    for (calc_abund_stats, calc_ani_ci) in test_cases {
+        let test_name = format!(
+            "abund{}_ani_ci{}",
+            calc_abund_stats as u8, calc_ani_ci as u8
+        );
+        group.bench_function(&test_name, |b| {
+            b.iter(|| {
+                let _result = calculate_gather_stats(
+                    black_box(&orig_query),
+                    black_box(&query),
+                    black_box(match_sig.clone()),
+                    black_box(&match_mh),
+                    black_box(42),  // Example match_size
+                    black_box(100), // Example remaining_hashes
+                    black_box(1),   // Example gather_result_rank
+                    black_box(200),
+                    black_box(calc_abund_stats),
+                    black_box(calc_ani_ci),
+                );
```

Comment:
`Result` is assigned but not checked here, so not necessarily doing any calculation. Replace with

```suggestion
                calculate_gather_stats(
                    black_box(&orig_query),
                    black_box(&query),
                    black_box(match_sig.clone()),
                    black_box(&match_mh),
                    black_box(42),  // Example match_size
                    black_box(100), // Example remaining_hashes
                    black_box(1),   // Example gather_result_rank
                    black_box(200),
                    black_box(calc_abund_stats),
                    black_box(calc_ani_ci),
                ).expect("error calculating gather stats");
```

(now it errors out :sweat_smile:
```
Benchmarking gather_stats/abund1_ani_ci0: Warming up for 3.0000 sthread 'main' panicked at src/core/benches/gather.rs:58:18:
error calculating gather: NeedsAbundanceTracking
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
)